### ISSUE TITLE
test: fix and unskip nosplit.go test

### ISSUE THREAD

**Issue by cherrymui** [golang/go#18317]
### What version of Go are you using (`go version`)?
tip

`test/nosplit.go` checks whether we have frame pointer enabled and skips the test if so.
```
	if strings.Contains(string(version), "framepointer") {
		// Skip this test if GOEXPERIMENT=framepointer
		return
	}
```

On non-AMD64 architectures, `GOEXPERIMENT=framepointer` is a no-op but the version string it prints always has `X:framepointer` *set*. So this test seems actually not run...

**Comment by minux** [comment]
And I think the long term plan is to have framepointer in all architectures?
having framepointers helped a lot in debugging and will help even more
for the RISC architectures.

**Comment by bradfitz** [comment]
Demoting to "maybe", but @cherrymui feel free to fix the test for Go 1.9 if you'd like still.

**Comment by rsc** [comment]
It shouldn't be skipped on amd64 either: it should be fixed everywhere.
